sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -93,6 +93,23 @@ class SparkSqlAstBuilder extends AstBuilder {
   }
 
   /**
+   * A command for users to list the properties for a table.
+   * If propertyKey is specified, the value for the propertyKey
+   * is returned. If propertyKey is not specified, all the keys
+   * and their corresponding values are returned.
+   * The syntax of using this command in SQL is:
+   * {{{
+   *   SHOW TBLPROPERTIES table_name[('propertyKey')];
+   * }}}
+   */
+  override def visitShowTblProperties(
+      ctx: ShowTblPropertiesContext): LogicalPlan = withOrigin(ctx) {
+    ShowTablePropertiesCommand(
+      visitTableIdentifier(ctx.tableIdentifier),
+      Option(ctx.key).map(_.STRING).map(string))
```

Comment:
This will only work STRING keys, whereas a partition key can also be an identifier. See: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala#L223-L229

Lets move parsing a `tablePropertyKey` into a seperate method (override `visitTablePropertyKey`) and use this for both cases.